src/vivarium/framework/population/population_view.py

```diff
@@ -79,9 +84,9 @@ def columns(self) -> list[str]:
         should be only be used in situations where the full state table is
         actually needed, like for some metrics collection applications.
         """
-        if not self._columns:
-            return list(self._manager.get_population(True).columns)
-        return list(self._columns)
+        if self.requires_all_columns:
+            return list(self._manager.get_population(True).columns) + self._columns
+        return self._columns
```

Comment:
If you access this property after the initialization phase of the simulation, are the created columns duplicated? I suspect they are. If so, you should use sets to weed out the duplicates.